Chaincase.Common/Services/AesThenHmac.cs

```diff
@@ -0,0 +1,402 @@
+﻿/*
+ * based on Modern Encryption of a String C#, by James Tuley, 
+ * https://gist.github.com/4336842
+ * 
+ * Dan Gould 2021
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Chaincase.Common.Services
+{
+    public static class AesThenHmac
+    {
+        private static readonly RandomNumberGenerator Random = RandomNumberGenerator.Create();
+
+        //Preconfigured Encryption Parameters
+        public static readonly int BlockBitSize = 128;
+        public static readonly int KeyBitSize = 256;
+
+        //Preconfigured Password Key Derivation Parameters
+        public static readonly int SaltBitSize = 64;
+        public static readonly int Iterations = 10000;
```

Comment:
how long does this take to run on a macbook? 10k iterations for pbkdf2 seems kind alow

Reply:
looks like 1Password has been using 100k iterations since 2016 https://support.1password.com/pbkdf2/